primitives/src/lib.rs

```diff
@@ -19,20 +19,24 @@ mod app {
 pub type AuthorityId = app::Public;
 
 sp_api::decl_runtime_apis! {
-    pub trait AlephApi {
-        fn authorities() -> Vec<AuthorityId>;
+    pub trait AlephApi<Id, BlockNumber>
+    where
+        Id: Encode + Decode,
+        BlockNumber: Encode + Decode + AtLeast32BitUnsigned,
+    {
+        fn current_session() -> Option<Session<Id, BlockNumber>>;
+        fn next_session() -> Option<Session<Id, BlockNumber>>;
     }
 }
 
 #[derive(Decode, Encode, PartialEq, Eq, Clone)]
-pub enum AuthoritiesLog<Id, Number>
+pub struct Session<Id, BlockNumber>
 where
     Id: Encode + Decode,
-    Number: Encode + Decode,
+    BlockNumber: Encode + Decode + AtLeast32BitUnsigned,
```

Comment:
Is it necessary for BlockNumber to be ` AtLeast32BitUnsigned`?

Reply:
It is not. I removed it